### PR TITLE
MG 232 - Refactoring immunohisto_transform to reuse it in transform_model_details

### PR DESCRIPTION
**Pull request by beatrizsaldana**
## Problem
The `immunohisto_transform` function was originally designed as a simple grouping utility that lacked proper data validation, standardization, and reusability. The `model_details` transform needed to process biomarker and pathology data with specific transformations (column renaming, data standardization, beta symbol replacement, etc.), but the original `immunohisto_transform` didn't provide these capabilities. This led to code duplication and inconsistent data processing between different transforms.

## Solution
Refactored the `immunohisto_transform` to be a comprehensive, reusable transformation module that can be shared between the standalone immunohisto transform and the `model_details` transform. Key improvements include:

1. **Enhanced Data Processing**: Added a new `prepare_immunohisto_data()` function that handles:
   - Missing value handling (fillna with empty strings)
   - Text standardization (capitalizing sex and tissue values)
   - Beta symbol replacement (`beta` → `&beta;`)
   - Column renaming (`type` → `evidence_type`, `measurement` → `value`, `age_death` → `age`)
   - Age formatting (appending "months" to age values)

2. **Improved Validation**: 
   - Added `REQUIRED_INPUT` constant to define expected columns for biomarkers and pathology datasets
   - Integrated `check_required_datasets_and_columns()` utility for comprehensive validation
   - Added validation to ensure at least one of "biomarkers" or "pathology" datasets is present

3. **Integration with model_details**: 
   - The `model_details` transform now uses `immunohisto_transform` to process both biomarkers and pathology data consistently
   - Added comprehensive test coverage for both transforms

## Test
- **Unit Tests**: Added extensive test coverage for the new `prepare_immunohisto_data()` function with various edge cases (empty values, None values, etc.)
- **Integration Tests**: Updated existing immunohisto transform tests to work with the new one and added comprehensive test assets
- **Data Consistency**: Verified that both transforms produce consistent, standardized output formats

The refactoring maintains backward compatibility for existing immunohisto transform usage while providing enhanced functionality for the new model_details transform, ensuring consistent data processing across the Model AD project.

### CODE REVIEW

File: src/agoradatatools/etl/transform/immunohisto_transform.py [hunk 1]
```diff
@@ -4,18 +4,86 @@
...
+    df = df.fillna("")
+    df["sex"] = df["sex"].str.title()
+    df["tissue"] = df["tissue"].str.title()
+
+    # Replace 'beta' with '&beta;' in biomarker types
+    df["type"] = df["type"].str.replace("beta", "&beta;")
+
+    # Rename columns
+    df = df.rename(
+        columns={"type": "evidence_type", "measurement": "value", "age_death": "age"}
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/immunohisto_transform.py]: Minor nitpick: is this something that can be done as a `column_rename` in the config file instead?

File: tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv [hunk 2]
```diff
@@ -1,5 +1,5 @@
-model,type,measurement,units,age_death,tissue,sex,genotype
-ModelA,TypeA,1,A,1,TissueA,male,genotype1
-ModelA,TypeA,1,A,1,TissueA,male,genotype1
-ModelA,TypeA,1,A,1,TissueA,male,genotype2
-ModelA,TypeA,1,A,1,TissueA,male,genotype2
+model,type,measurement,units,age_death,tissue,sex,genotype,individual_id
```
Comment (jaclynbeck-sage) [inline tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: I don't think I noticed this before, but both the pathology and biomarkers file have an `individualID` and a `specimenID` column. Is there a reason that `specimenID` is missing in these test files?

File: src/agoradatatools/etl/transform/immunohisto_transform.py [hunk 3]
```diff
@@ -24,18 +92,28 @@ def immunohisto_transform(
...
         dataset_name (str): The name of the dataset to transform.
-        group_columns (List[str], optional): List of columns to group by. Defaults to ['model', 'type', 'age_death', 'tissue', 'units'].
-        extra_columns (List[str], optional): List of columns to include in the group. Defaults to ['genotype', 'measurement', 'sex'].
-        extra_column_name (str, optional): Name of the column containing the extra columns. Defaults to 'points'.
+        group_columns (List[str], optional): List of columns to group by. Defaults to ['model', 'evidence_type', 'tissue', 'age', 'units'].
+        extra_columns (List[str], optional): List of columns to include in the group. Defaults to ['genotype', 'sex', 'individual_id', 'value'].
+        extra_column_name (str, optional): Name of the column containing the extra columns. Defaults to 'data'.
 
     Returns:
         pd.DataFrame: A DataFrame grouped by the group_columns.
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/immunohisto_transform.py]: Could you add a little more information to both the Returns comment and the initial description about the structure of the output data frame? I think something like: "A DataFrame containing all group_columns, plus an extra column named as specified in extra_column_name. This extra column contains all information from extra_columns, collapsed into a single dictionary."

File: src/agoradatatools/etl/transform/immunohisto_transform.py [hunk 4]
```diff
@@ -46,4 +124,6 @@ def immunohisto_transform(
         entry[extra_column_name] = group[extra_columns].to_dict("records")
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/immunohisto_transform.py:153]: I think because I don't use `zip` or grouping very much in Python, I'm having a little trouble parsing what this `for` loop does. It would be really helpful if you could add a comment or two in/above the for loop describing it.

**Review by jaclynbeck-sage — CHANGES_REQUESTED** [review]
I love when a good refactor deletes a bunch of code! This is great, I just have a few minor requests which are mostly to add some more comments. I really appreciate the error checking you are adding to the code base, too.

File: src/agoradatatools/etl/transform/immunohisto_transform.py (hunk 1 above)
Comment (thomasyu888) [reply src/agoradatatools/etl/transform/immunohisto_transform.py]: There is this bug that we are going to investigate: https://sagebionetworks.jira.com/browse/DPE-1384

Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/immunohisto_transform.py]: Oh interesting! However we can still use `column_rename` for this one.

File: tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv (hunk 2 above)
Comment (JessterB) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: We don't plan to display specimenID in the explorer, so I didn't specify it in the target payload. Omitting it might save a very, very small amount of time and resources!

Comment (JessterB) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: @jaclynbeck-sage Do you think users would want to see the specimenID? My assumption was that once the individual data was joined to the results, no one would care about it. It wouldn't be difficult to leave it in and surface it along with the individualID (in box plot dot tooltips) though.

File: src/agoradatatools/etl/transform/immunohisto_transform.py (hunk 1 above)
Comment (beatrizsaldana) [reply src/agoradatatools/etl/transform/immunohisto_transform.py]: Done!

File: tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv (hunk 2 above)
Comment (beatrizsaldana) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: Let me know how we want to handle this and I'll add specimenID if we need it :)

Comment (jaclynbeck-sage) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: Now that I think on it, I don't think the specimen ID is very important for the explorer. Especially because I don't think they did replicates for the same individual so it's probably a 1:1 mapping. So I think it's fine to leave them out. Not sure I'd really thought on it when I made the first comment!

Comment (jaclynbeck-sage) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: Oh wait, I'm confusing myself. I think the reason I wanted `specimenID` in the test files was so that the test files had the same columns and format as the real input files, in the same order. Right now, if the test data is missing things the real data has, we are not actually testing how the function performs on real input. We don't need it in the data output for the explorer, but I think I wanted the test input to be as close to the real input as possible.

Comment (JessterB) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: Oooh if this is for adding specimenId to the test files, then this one's for you @beatrizsaldana

Comment (beatrizsaldana) [reply tests/test_assets/immunohisto_transform/input/immunohisto_transform_duplicated_input.csv]: @jaclynbeck-sage I added the specimen_id column to the testing files so the PR is ready for re-review :)

**Review by jaclynbeck-sage — APPROVED** [review]
Excellent, thank you for adding the specimen ID! Approved.
